src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,320 @@
+        let logger                = Logger::new("ProjectName");
+        let display_object        = display::object::Instance::new(&logger);
+        let font                  = scene.fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let size                  = Vector2::new(scene.camera().screen().width,TEXT_SIZE);
+        let base_color            = TRANSPARENT_TEXT_COLOR;
+        let text_size             = TEXT_SIZE;
+        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field            = TextField::new(scene,text_field_properties,focus_manager);
+        let view_logger           = Logger::sub(&logger,"view_logger");
+        let view                  = component::ShapeView::<background::Shape>::new(&view_logger, scene);
+        let project_name          = Rc::new(RefCell::new(UNKNOWN_PROJECT_NAME.to_string()));
+        let name_output           = frp.outputs.name.clone();
+        let animations            = Animations::new(&frp.network);
+        Self{logger,view,display_object,text_field,project_name,name_output,animations}.init()
+    }
+
+    fn update_center_alignment(&self) {
+        let width  = self.text_field.width_of_line(0);
+        let offset = Vector3::new(-width/2.0,0.0,0.0);
+        let height = self.text_field.line_height();
+        self.view.shape.sprite.size.set(Vector2::new(width,height));
+        self.view.set_position(Vector3::new(0.0,-height/2.0,0.0));
+        self.animations.position.set_target_value(offset);
+    }
+
+    fn init(self) -> Self {
+        //FIXME:Use add_child(&text_field) when replaced by TextField 2.0
+        self.add_child(&self.text_field.display_object());
+        self.add_child(&self.view);
+        let project_name = self.clone_ref();
+        //FIXME[dg]: This section to check newline and keep TextField in a single line is hacky
+        // and should be removed once the new TextField is implemented.
+        self.text_field.set_text_edit_callback(move |change| {
+            // If the text edit callback is called, the TextEdit must be still alive.
+            let field_content = project_name.text_field.get_content();
+            let new_name      = field_content.replace("\n","");
+            if change.inserted == "\n" {
+                project_name.rename(&new_name);
+            }
+            // Keep only one line.
+            project_name.text_field.set_content(&new_name);
+            project_name.update_center_alignment();
+        });
+
+        self.update_text_field_content();
+        self
+    }
+
+    fn reset_name(&self) {
+        info!(self.logger, "Resetting project name.");
+        self.update_text_field_content();
+    }
+
+    fn update_text_field_content(&self) {
+        self.text_field.set_content(&self.project_name.borrow());
+        self.update_center_alignment();
+    }
+
+    fn fade_in_text(&self) {
+        self.animations.opacity.set_target_value(1.0);
+    }
```

Comment:
We've been talking about that in reviews earlier - could you just use these calls directly in FRP instead of delegating them this way to model?